### ISSUE TITLE
Support language detection

### ISSUE THREAD

**Issue by eeejay** [eeejay/libspiel#4]
It would be cool if libspiel selected a language appropriate voice after determining automatically what language it is. This functionality could be added with a language detection library like CLD2 or something similar.

There would probably be a performance/latency cost so this should be an opt-in kind of feature. Maybe passing "auto" as language.

**Comment by TTWNO** [comment]
I think this massively expands the scope and complexity of a basic speech protocol. It would be

**Comment by eeejay** [comment]
Right, this would be in the client. It wouldn't add anything to the dbus spec or providers. Basic idea is that an app calls speak on some text and the client library does some automatic language detection and voice selection and then sends it to the appropriate provider.
